packages/libcap.rb

```diff
@@ -2,48 +2,55 @@
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libcap/2.49_x86_64/libcap-2.49-chromeos-x86_64.tar.xz'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libcap/2.62_armv7l/libcap-2.62-chromeos-armv7l.tpxz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libcap/2.62_armv7l/libcap-2.62-chromeos-armv7l.tpxz',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libcap/2.62_i686/libcap-2.62-chromeos-i686.tpxz',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libcap/2.62_x86_64/libcap-2.62-chromeos-x86_64.tpxz'
   })
   binary_sha256({
-    aarch64: 'a70c7ce2ce27a15a294ce7229c10a1c2227c2705a8b0e0be88cf8bb5eed73e77',
-     armv7l: 'a70c7ce2ce27a15a294ce7229c10a1c2227c2705a8b0e0be88cf8bb5eed73e77',
-       i686: 'ff41b92eeeb4c068086e67322b6a3fa9609b883a84c29f5a10c51dc3d38aa09c',
-     x86_64: 'bc72fba8169275a4e7a78ee90eea47a20b42fa0544eb321d2c1c1549640982b6'
+    aarch64: 'ad1ba9eeb6311aa50cae4933861e6ef3c6660114f8f59fae8b38ec33cf52f4ba',
+     armv7l: 'ad1ba9eeb6311aa50cae4933861e6ef3c6660114f8f59fae8b38ec33cf52f4ba',
+       i686: 'fde7113445a251b6109911cbd5de39300ec2469091cdaea6bc6a57e726e0ef86',
+     x86_64: '3964a327f72a3323e1a192d5fbe4020f29e83ede75347e0d65582ace36f4f66b'
   })
 
   depends_on 'gperf' => :build
   depends_on 'linux_pam'
-  depends_on 'patchelf' => :build
 
   def self.build
     # add includes option to make it work with gperf-3.1
     system 'sed -e "/gperf --/s/gperf --/gperf --includes --/" -e "/gperf --/s/cap_lookup_name(/cap_dummy(/" -i libcap/Makefile'
-
     # change the path to ld
     system "sed -i 's,/usr/bin/ld,#{CREW_PREFIX}/bin/ld,g' Make.Rules"
     # change prefix
-    system "sed -i 's,prefix=/usr,prefix=#{CREW_PREFIX},' Make.Rules"
+    system "sed -i 's,prefix=/usr,prefix=#{CREW_PREFIX},g' Make.Rules"
     # set exec_prefix
-    system "sed -i 's,^exec_prefix=,exec_prefix=\$(prefix),' Make.Rules"
+    system "sed -i 's,^exec_prefix=,exec_prefix=\$(prefix),g' Make.Rules"
+    # use system libdir
+    system "sed -i 's,^lib_prefix=\$(exec_prefix),lib_prefix=#{CREW_LIB_PREFIX},g' Make.Rules"
     # http://git.yoctoproject.org/cgit.cgi/poky/plain/meta/recipes-support/libcap/files/0001-ensure-the-XATTR_NAME_CAPS-is-defined-when-it-is-use.patch
     system 'sed -i "s,^\#ifdef VFS_CAP_U32,\#if defined (VFS_CAP_U32) \&\& defined (XATTR_NAME_CAPS),g" libcap/cap_file.c'
-    system "env CFLAGS='-flto=auto' CXXFLAGS='-flto=auto' \
-        LDFLAGS='-flto=auto' make"
+    system "#{CREW_ENV_OPTIONS} make"
   end
 
   def self.install
     system 'make', 'RAISE_SETFCAP=no', "DESTDIR=#{CREW_DEST_DIR}", 'install'
-    system "patchelf --set-rpath #{CREW_LIB_PREFIX} #{CREW_DEST_PREFIX}/sbin/capsh"
+  end
+
+  def self.check
+    @container_check = system 'grep :/docker /proc/self/cgroup &> /dev/null'
+    # Tests do not work in container.
+    return unless @container_check
```

Comment:
`return if @container_check`?  What is returned?  True or false?

Reply:
Do tests pass on actual hardware?

Reply:
> `return if @container_check`?  What is returned?  True or false?

`@container_check` is false if in a container.

Reply:
Actually that doesn't work. In docker the only reliable way I'm finding of determining if you are in a container is looking for `/.dockerenv` .

Reply:
Tests pass on `x86_64`, and checking now on `i686`... Do not have `armv7l` hardware to test on...

Reply:
I can check on arm.  What are you doing to confirm your test?

Reply:
Passes on `i686` too.

Reply:
That doesn't seem like a reliable way to check for a docker environment.  See https://tuhrig.de/how-to-know-you-are-inside-a-docker-container/.

Reply:
@supechicken FYI on container check in sommelier needing new logic...

Reply:
@uberhacker Can you check if the test also works on armv7l? I suspect it will...

Reply:
The method in your link doesn't work.

```
chronos@cheekon-x86_64 /proc/self $ cat /proc/1/cgroup
0::/
```

I used this method: https://stackoverflow.com/a/25518345

Reply:
On armv7l, I get:
```
make -C doc all
make[1]: Entering directory '/usr/local/tmp/crew/libcap.20211228234705.dir/doc'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/usr/local/tmp/crew/libcap.20211228234705.dir/doc'
Preconfiguring package...
/usr/local/lib/crew/lib/package.rb:195:in `system': No such file or directory - make -j4
libcap failed to build: `make -j4 RAISE_SETFCAP=no DESTDIR=/usr/local/tmp/crew/dest install` exited with 127
```
Same inside the container.  I think there is a problem with @supechicken's latest crew update.

Reply:
Yeah you're hitting https://github.com/skycocker/chromebrew/issues/6529

Reply:
That begs the question.  How have you been successfully generating the latest series of binaries?